middlewares/middlewares.go

```diff
@@ -73,7 +74,11 @@ func getRemoteAddr(r *http.Request) string {
 	if len(address) > 0 {
 		return address
 	}
+	if splitIndex := strings.Index(r.RemoteAddr, ":"); splitIndex > 0 {
```

Comment:
`-1` denotes the nonexistence of a substring in `strings.Index`. If for some reason `:` is the first character in the string, this will still return `:8907`. It's _probably_ okay though.